editor/components/block-list/block.js

```diff
@@ -398,6 +409,19 @@ export class BlockListBlock extends Component {
 		}
 	}
 
+	insertTemplateBlocksIfInnerBlocksEmpty( template ) {
+		const { block, onInsertBlocks, uid } = this.props;
+		if ( template && ! block.innerBlocks.length ) {
+			onInsertBlocks( synchronizeBlocksWithTemplate( [], template ), 0, uid );
```

Comment:
I think synchronizing the template is not needed here because we only synchronize templates if the template is locked but we don't support locking yet for inner Blocks.

Unless I'm missing something?

Reply:
Yes synchronizing the template is not needed. But I want to given a template parse and create the block structure it represents turns out that's exactly what `synchronizeBlocksWithTemplate( [], template )` does so I did not wanted to duplicate code and used it :)